### DISCUSSION TITLE
Is there a way to track whether fields are dirty after the default values have been reset?

### DISCUSSION THREAD

**Discussion by ErnestGrey** [react-hook-form/react-hook-form#6860] [Q&A]
I have component that contains a form that I use for both creating and editing a team. When the component is in "edit" mode I set the default values initially and then in my `useEffect` I fetch the team data and then use `reset()` to overwrite the initial default values so the "edit" mode has the form pre-filled. I have a button that I disable if the the component `isEditing && !isDirty`. Initially this works because if any of the fields change the button is then enabled, but if I change a field back to the reset value `isDirty` is still true.

Here's an example for `name` field:

1. `defaultValue`: ""
2. value of `name` after `reset` is called in `useEffect`: "Red Team"
3. after `reset` called: `isDirty === false`
4. change `name` from "Red Team" to "Blue Team" and `isDirty === true`
5. change `name` from "Blue Team" back to "Red Team" and `isDirty === true`

Is there a way for `isDirty` to be `false` on the last step? Or is there a different way to check against the reset values?

**Comment by JamesWong1999** [comment]
can you share a codesandbox?

**Reply by ErnestGrey** [reply]
Hey James, here is a [link](https://codesandbox.io/s/practical-frog-zux70?file=/src/App.js) to a dumbed down version of what I am doing.

Two things:

1. It is working as it should in the sandbox, but not in my app (I also have about 5 other inputs in the form), maybe I need to comment out all of the inputs and go one by one through them?
2. Getting the `Warning: Can't perform a React state update on an unmounted component...` error as well and not sure why.
